server/src/main/java/org/elasticsearch/action/update/TransportUpdateAction.java

```diff
@@ -191,21 +191,59 @@ protected void shardOperation(final UpdateRequest request, final ActionListener<
                 IndexRequest upsertRequest = result.action();
                 // we fetch it from the index request so we don't generate the bytes twice, its already done in the index request
                 final BytesReference upsertSourceBytes = upsertRequest.source();
-                client.bulk(toSingleItemBulkRequest(upsertRequest), wrapBulkResponse(ActionListener.<IndexResponse>wrap(response -> {
-                    UpdateResponse update = new UpdateResponse(
-                        response.getShardInfo(),
-                        response.getShardId(),
-                        response.getId(),
-                        response.getSeqNo(),
-                        response.getPrimaryTerm(),
-                        response.getVersion(),
-                        response.getResult()
-                    );
-                    if (request.fetchSource() != null && request.fetchSource().fetchSource()) {
-                        Tuple<XContentType, Map<String, Object>> sourceAndContent = XContentHelper.convertToMap(
-                            upsertSourceBytes,
-                            true,
-                            upsertRequest.getContentType()
+                client.bulk(
+                    BulkRequest.fromSingleRequest(upsertRequest),
+                    TransportBulkAction.wrapBulkAsSingleItemResponse(ActionListener.<IndexResponse>wrap(response -> {
```

Comment:
Can we have these methods as unqualified static imports like the existing code does? I don't have strong preference for this style. But in this particular case, it should cause less cascading style changes and be easier for review.

Reply:
Yup I can do this. I went with this format because `fromSingleRequest` without the class is pretty awkward but don't feel too strongly.

Reply:
😬 sorry, lines of code length is not on our side so there's still a bit of cascading formatting diff that happens

Reply:
No worries. Thanks for going through the trouble of changing the imports.